service/src/test/java/greencity/service/EventServiceImplTest.java

```diff
@@ -971,6 +971,36 @@ void getEventsForAuthorizedUserTest() {
         verify(eventRepo).loadEventDataByIds(idsPage.getContent(), userId);
     }
 
+    @Test
+    void testGetEventsForAuthorizedUserExceedingPageTest() {
+        int requestedPage = 10;
+        int totalPages = 2;
+        int pageSize = 6;
+        Pageable pageable = PageRequest.of(requestedPage, pageSize);
+        Long userId = 1L;
+        FilterEventDto filterEventDto = getFilterEventDto();
+
+        Page<Long> idsPage = new PageImpl<>(
+            List.of(3L, 1L),
+            PageRequest.of(0, pageSize),
+            totalPages * pageSize);
+
+        when(restClient.findById(userId)).thenReturn(getUserVO());
+        when(eventRepo.findEventsIds(pageable, filterEventDto, userId)).thenReturn(idsPage);
+
+        BadRequestException exception = assertThrows(
+            BadRequestException.class,
+            () -> eventService.getEvents(pageable, filterEventDto, userId),
+            "Expected BadRequestException to be thrown");
+
+        String expectedMessage = String.format("Requested page %d exceeds total pages %d.", requestedPage, totalPages);
```

Comment:
If you already have PAGE_NOT_FOUND_MESSAGE, you can use it for expectedMessage in the test instead of writing it again